src/MySqlConnector/Core/CachedProcedure.cs

```diff
@@ -194,6 +194,9 @@ internal static List<CachedParameter> ParseParameters(string parametersSql)
 
 		internal static string ParseDataType(string sql, out bool unsigned, out int length)
 		{
+			// remove trailing "NULL" which is needed for SingleStore compatibility
+			sql = Regex.Replace(sql, @"\s+NULL\s*$", "", RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
```

Comment:
I suppose there might be issues with this. `NULL` and `NOT NULL` are part of the type in the SingleStore engine. If you declare a parameter as `NOT NULL` will produce a string like `paramName INT NOT`.

Reply:
```
			if (sql.EndsWith(" NOT NULL", StringComparison.OrdinalIgnoreCase))
				sql = sql.Substring(0, sql.Length - 9);
			if (sql.EndsWith(" NULL", StringComparison.OrdinalIgnoreCase))
				sql = sql.Substring(0, sql.Length - 5);
```
Can be like that, but it might make sense to create an output boolean parameter isNullable.